PLAT-71: Truncate the task role name_prefix is len() > 32

keymon opened the pull request:
In order to avoid an error like this:

* module.lambda_s3_backup.module.s3_backup_taskdef.aws_iam_role.task_role:
  "name_prefix" cannot be longer than 32 characters, name is limited to 64

We replace the name prefix with a truncated + a hash version.

keymon committed:
Truncate the task role name_prefix is len() > 32

In order to avoid an error like this:

* module.lambda_s3_backup.module.s3_backup_taskdef.aws_iam_role.task_role:
  "name_prefix" cannot be longer than 32 characters, name is limited to 64

We replace the name prefix with a truncated + a hash version.

bart613 reviewed (changes requested):
Can we have a test to cover this case?

keymon commented:
> Can we have a test to cover this case?

Sure!

keymon committed:
Fix role test and remove spurious prints

When testing the role, the policy to match must be for the Principal
ecs-tasks.amazonaws.com as per the PR #2[1]

We also remove unnecessary prints

[1] https://github.com/mergermarket/tf_ecs_task_definition_with_task_role/pull/2

keymon committed:
Add test case for long family name (>32 chars)

We want to test that the creation of the role will not fail when the
family name passed to the module is >32.

Error to avoid:

    * module.taskdef_with_role.aws_iam_role.task_role: "name_prefix" cannot be longer than 32 characters, name is limited to 64

keymon commented:
@bart613 done
